server/public/model/config.go

```diff
@@ -3197,6 +3197,38 @@ func (s *PluginSettings) SetDefaults(ls LogSettings) {
 	}
 }
 
+// Sanitize cleans up the plugin settings by removing any sensitive information.
+// It does so by checking if the setting is marked as secret in the plugin manifest.
+// If it is, the setting is replaced with a fake value.
+// If a plugin is no longer installed, all settings of it's are sanitized.
+// If the list of manifests in nil, i.e. plugins are disabled, all settings are sanitized.
+func (s *PluginSettings) Sanitize(pluginManifests []*Manifest) {
+	manifestMap := make(map[string]*Manifest, len(pluginManifests))
+
+	for _, manifest := range pluginManifests {
+		manifestMap[manifest.Id] = manifest
+	}
+
+	for id, settings := range s.Plugins {
+		manifest := manifestMap[id]
+
+		for key := range settings {
+			if manifest == nil {
+				// Sanitize plugin settings for plugins that are not installed
+				settings[key] = FakeSetting
+				continue
+			}
```

Comment:
@hanzei I've just noticed this may be causing somewhat of a breaking change that, in combination with some less than ideal logic, is currently making Calls e2e tests fail completely. 

In our case some config fields [were set to empty](https://github.com/mattermost/mattermost-plugin-calls/blob/3965559e56f47db165ff7afc283f213a5e2a94ec/e2e/overlay-config.jq#L20-L23) originally and they would result in automatic sanitization to FakeSetting (a bunch of asterisks) to kick in which with the way we [patch the config](https://github.com/mattermost/mattermost-plugin-calls/blob/d0af50c78ccedb286337b379a064a3c80293fe0e/e2e/global-setup.ts#L214-L227) would cause to actually set them as real values, resulting in all sorts of issues.

It may very well be an edge case but wanted to bring it up since I was very confused when debugging this issue.

Reply:
That is indeed a breaking change. As discussed earlier in the PR, I think it's worth doing that change to avoid leaking secrets via the API.

One alternative (that is still a breaking change) would be to omit the plugin configuration from plugins that are not installed completely from the response instead of obfuscating them.

Would that make the API easier to use?

One question regarding the Calls E2E tests: Does it make sense to install the Calls plugin first before making the preparations like patching the config?

Reply:
@hanzei I was finally able to reproduce this issue locally. The problem wasn't about patching the config. In our e2e pipeline the step before installing Calls is installing Playbooks which somehow results in committing the fake values to the config store for any plugin not yet installed but that has some config fields set. I think we should prevent this side effect from happening.

Reply:
Submitted https://github.com/mattermost/mattermost/pull/28260